### PR TITLE
sql: add doc for tidb executor

### PR DESCRIPTION
**Pull request by XuHuaiyu**
This PR does 2 things:
- [ ] add the explanation for Join Algorithms (working on drawing elementary diagram, commit will be pushed soon)
- [x] add the explanation for Aggregate Algorithms

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,9 +18,10 @@
     - [TiDB 访问权限管理](sql/privilege.md)
     - [TiDB 用户账户管理](sql/user-account-management.md)
     - [使用加密连接](sql/encrypted-connections.md)
-  + SQL 优化
+  + SQL 优化与执行
```
Comment (shenli) [inline README.md]: s/优化与执行/优化器与执行引擎/

File: sql/executor.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+| 2  4     | [1] (2, 3)     [2] (3, 3)  |
+
+输入数据输入完后，扫描 Hash 表并计算，便可以得到最终结果：
+
+| Hash 表    | avg(b) |
+| ---------- | ------ |
+| [1] (2, 3) | 2      |
+| [2] (1, 3) | 1      |
+
+TiDB 中，为了加快执行速度，Hash Aggregate 在聚合函数参数不为 distinct 时并行执行。
```
Comment (zz-jason) [inline sql/executor.md]: 前面那个简单版本的一句话就可以讲完，后面这个并行执行的模式可以深入讲讲。。

Comment (zz-jason) [reply sql/executor.md]: 比如可以介绍一下那几种聚合的状态，不同状态的输入输出是什么，状态间如何交互的等等

**Comment by XuHuaiyu** [comment]
Paused. I'll reopen this later.

**Pull request closed without merging**